subiquity/ui/views/filesystem/filesystem.py

```diff
@@ -274,12 +278,13 @@ def _disk_REMOVE(self, disk):
         disk._constructed_device = None
         self.parent.refresh_model_inputs()
 
-    def _disk_TOGGLE_BOOT(self, disk):
+    async def _disk_TOGGLE_BOOT(self, disk):
         if boot.is_boot_device(disk):
-            self.parent.controller.remove_boot_disk(disk)
+            await self.parent.controller.remove_boot_disk(disk)
+            self.parent.refresh_model_inputs()
         else:
-            self.parent.controller.add_boot_disk(disk)
-        self.parent.refresh_model_inputs()
+            await self.parent.controller.add_boot_disk(disk)
+            self.parent.refresh_model_inputs()
```

Comment:
nit: is there any reason for adding `self.parent.refresh_model_inputs()` to both clauses?